src/ImageSharp/Processing/Processors/Convolution/BorderWrappingMode.cs

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+namespace SixLabors.ImageSharp.Processing.Processors.Convolution
+{
+    /// <summary>
+    /// Wrapping mode for the border pixels in convolution processing.
+    /// </summary>
+    public enum BorderWrappingMode : byte
+    {
+        /// <summary>Repeat the border pixel value: aaaaaa|abcdefgh|hhhhhhh</summary>
+        Repeat = 0,
+
+        /// <summary>Take values from the opposite edge: cdefgh|abcdefgh|abcdefg</summary>
+        Wrap = 1,
+
+        /// <summary>Mirror the last few border values: fedcb|abcdefgh|gfedcb</summary>
+        /// <remarks>Please note this mode does not repeat the very border pixel, as this gives better image quality.</remarks>
+        Mirror = 2
```

Comment:
Does this negatively affect the tiling of convoluted images?

Reply:
Not sure if I understand you correctly, but for image tiles, you would use Wrap mode.

To my knowledge Mirror is the most natural mode for photos. If you would repeat the border pixel (a and h in the example) the image might get a small area of equal pixel value, which looks unnatural.

However, for Graphics, especially with solid background, Clamp is the best choice.

Having said that, applying the wrong mode for the subject can have awkward effects.

Reply:
Yep, sorry, that makes sense. 

Let's make sure we use Clamp over Repeat for naming to match CUDA definitions. 

https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__TYPES.html#group__CUDART__TYPES_1g03e1bbd2c491d809279c7a47e2cd0351

It appears to me that they repeat the border pixel though (I may be reading this wrong)

https://stackoverflow.com/a/27411685/427899

Reply:
The names now match with CUDA's, with the deviation you mention indeed. 

Alternatively, [OpenCV](https://vovkos.github.io/doxyrest-showcase/opencv/sphinx_rtd_theme/enum_cv_BorderTypes.html) has both Mirror modes (which it calls `Reflect` and `Reflect_101`). OpenCV does take the Mirror mode without duplicating the border pixel as the default.

I bet you, if you find a third library, it has its border enum implemented even differently.

I can easily make an extra Mirror mode with the border pixel repeated. Main difficulty there is coming up with a good name for the 2 variants of Mirror.

Reply:
Ooh YES! I love it!

Reply:
Bounce mode is in

Reply:
Ace. Will review ASAP. Thanks for your patience.

Reply:
Also, please advise how to expose the `BorderWrappingMode` and in which ImageProcessor (or any other public API).

Reply:
OK. 

- I don't think the functionality should be added to Resize. 
- We should add both X and Y properties as overloads to our Gaussian Blur/Sharpen methods and Box Blur extensions. I don't think the Bokeh blur functionality can be augmented in that fashion.

Reply:
Ok, done as stated above